### DISCUSSION TITLE
ScriptReference/AssetImporters.ScriptedImporter

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#680] [Notes]
# ScriptReference/AssetImporters.ScriptedImporter

https://docs.unity3d.com/ScriptReference/AssetImporters.ScriptedImporter.html

**Comment by FreyaHolmer** [comment]
**Do not set the `name` of the main asset to something unique per-asset** - if your main asset is a prefab GameObject at least, it's an identifier similar to the first parameter in `AssetImportContext.AddObjectToAsset(...)`, as far as I can tell, so **it has to be the same across all assets**, otherwise references will break, and you will get weird behavior with your prefab instances silently resetting to 0,0,0 in your scenes

Unity will automatically set the name of the main asset prefab to the name of the file regardless, so don't try and do that manually like I did c:

[Related thread on this issue](https://discussions.unity.com/t/issue-with-retaining-fileids-in-unitys-scripted-importer/925024)
